spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/actions/SnapshotTableSparkAction.java

```diff
@@ -124,10 +126,10 @@ private SnapshotTable.Result doExecute() {
     StagedSparkTable stagedTable = stageDestTable();
     Table icebergTable = stagedTable.table();
 
-    // TODO: Check the dest table location does not overlap with the source table location
```

Comment:
@RussellSpitzer I believe you added the original TODO. Does this change address it as intended? We now check for location overlap using normalized absolute Paths and reject equal/ancestor/descendant cases.

Reply:
@RussellSpitzer Sorry for the confusion. My question was whether this PR fullfills the TODO.
This PR has added a check for location overlap using normalized absolute Paths and rejected equal/ancestor/descendant cases. Does it satisfies the TODO?

Reply:
@huaxingao thank you very much for reviewing the code! @RussellSpitzer's statement is accurate, and I have found the relevant validation code. Apologies for previously focusing only on the todo and overlooking the existing code details.

https://github.com/apache/iceberg/blob/170189e17ca0fd423e690d0de52380fa092177fd/spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/actions/SnapshotTableSparkAction.java#L217-L223

Should I have the opportunity to update the SnapShot code in the future, I will remove this todo.